operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -1,52 +1,55 @@
 package io.javaoperatorsdk.operator;
 
-import io.fabric8.kubernetes.api.model.apiextensions.v1.CustomResourceDefinition;
+import java.io.IOException;
+import java.net.ConnectException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import io.fabric8.kubernetes.client.CustomResource;
 import io.fabric8.kubernetes.client.KubernetesClient;
 import io.fabric8.kubernetes.client.Version;
 import io.javaoperatorsdk.operator.api.ResourceController;
 import io.javaoperatorsdk.operator.api.config.ConfigurationService;
 import io.javaoperatorsdk.operator.api.config.ControllerConfiguration;
-import io.javaoperatorsdk.operator.processing.event.DefaultEventSourceManager;
-import java.io.Closeable;
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Objects;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import io.javaoperatorsdk.operator.processing.ConfiguredController;
+import io.javaoperatorsdk.operator.processing.DefaultEventHandler;
+import io.javaoperatorsdk.operator.processing.DefaultEventHandler.EventMonitor;
+import io.javaoperatorsdk.operator.processing.event.Event;
 
 @SuppressWarnings("rawtypes")
 public class Operator implements AutoCloseable {
   private static final Logger log = LoggerFactory.getLogger(Operator.class);
   private final KubernetesClient k8sClient;
   private final ConfigurationService configurationService;
-  private final List<Closeable> closeables;
-  private final Object lock;
-  private final List<ControllerRef> controllers;
-  private volatile boolean started;
-  private final Metrics metrics;
+  private final ReentrantLock lock = new ReentrantLock();
+  private final List<ConfiguredController> controllers = new LinkedList<>();
+  private volatile boolean started = false;
 
-  public Operator(
-      KubernetesClient k8sClient, ConfigurationService configurationService, Metrics metrics) {
+  public Operator(KubernetesClient k8sClient, ConfigurationService configurationService) {
     this.k8sClient = k8sClient;
     this.configurationService = configurationService;
-    this.closeables = new ArrayList<>();
-    this.lock = new Object();
-    this.controllers = new ArrayList<>();
-    this.started = false;
-    this.metrics = metrics;
+    DefaultEventHandler.setEventMonitor(new EventMonitor() {
```

Comment:
I think this would be better to have these metrics per controller. (Something to think about in the future)

Reply:
Might make sense, indeed, though we can side-step the issue by adding the controller's name to the counter tags so that we can only have one handler and still be able to filter per controller. Having a different monitor per controller would only make sense if we wanted to track different metrics per controller (which we indeed might).